files-community/Files
# Code Quality: Introduce details layout columns generator phase 1

## Summary

As of now, you have to add literally 250 lines only for one column when you are adding a new one (you can see [git column introduction PR](https://github.com/files-community/Files/pull/12485)). We are introducing columns generator to mitigate that situation.

## Task Checklist

- [ ] Introduce details layout columns generator

## Steps To Validate Changes

1. Launch the app
2. Go to a folder
3. Resize some columns
4. Hide some columns
5. Go to another folder
6. See the columns you hid are shown in that folder
7. Go to git folder
8. See git columns appear

If nothing happen, you confirmed layout settings for each page are stored correctly in user_settings.db, hide/show operation is working properly, the app is able to fetch settings from the database and use them in an actual layout.

## PR Checklist

- [x] **Development**: Related: #12505

## Screenshots

Related codebase structure:
![image](https://github.com/files-community/Files/assets/62196528/88e4eb54-9250-4e45-a996-dc4b0ef90151)

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.
